File: cscs-checks/prgenv/src/mpi/hello_world_mpi.f90
program mpihel
implicit none

include 'mpif.h'

    integer :: rank, size, ierr
    integer :: mpiversion, mpisubversion
    integer :: resultlen = -1
    character(len=MPI_MAX_LIBRARY_VERSION_STRING) :: mpilibversion

    call MPI_INIT(ierr)
    call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)
    call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierr)
    write (*,'(A23,1X,I3,1X,A6,1X,I3,1X,A12,1X,I3,1X,A6,I3)') 'Hello World from thread', 0, &
     'out of', 1, 'from process', rank, 'out of', size

    call MPI_Get_version(mpiversion, mpisubversion, iErr)
    call MPI_Get_library_version(mpilibversion, resultlen, ierr)
    print *, '# MPI-',mpiversion,'.',mpisubversion,' = ',trim(mpilibversion)
    !flush(6)

    call MPI_FINALIZE(ierr)

end program mpihel

File: cscs-checks/prgenv/src/mpi/hello_world_mpi.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char *argv[])
{
   int rank, size;
   int mpiversion, mpisubversion;
   int resultlen = -1;
   char mpilibversion[MPI_MAX_LIBRARY_VERSION_STRING];
   MPI_Init(&argc, &argv);
   MPI_Comm_rank(MPI_COMM_WORLD, &rank);
   MPI_Comm_size(MPI_COMM_WORLD, &size);
   printf("Hello, World from thread %d out of %d from process %d out of %d\n",
       0, 1, rank, size);

   MPI_Get_version( &mpiversion, &mpisubversion );
   MPI_Get_library_version(mpilibversion, &resultlen);
   printf( "# MPI-%d.%d = %s\n", mpiversion, mpisubversion, mpilibversion);

   MPI_Finalize();
   return 0;
}
